python/_tskitmodule.c

```diff
@@ -5226,6 +5226,11 @@ TableCollection_get_indexes(TableCollection *self, void *closure)
         goto out;
     }
 
+    if (!tsk_table_collection_has_index(self->tables, 0)) {
```

Comment:
I don't know about this. I think it would be logical to return an empty dictionary. That way we're more robust to adding more indexes in the future, where the dictionary may contain the current edge_x indexes, but also maybe any future indexes. So, we might have to deal with the situation where the dictionary contains a variable number of keys - logically then, it makes sense to treat the zero-keys situation in the same way.

Think of ``tsk_table_collection_has_index(self, 0)`` as saying "do you have all possible indexes". So, we'd have (roughly)
```
dict = PyDict_New();
if (tsk_table_collection_has_index(self->tables, 0)) {
    // add key-value pairs for each edge_insertion and edge_removal
}
```
We can see how this is then extensible to having more indexes in the future - we can just test the existance of the each index on a case-by-case basis.

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/921/commits/2855dec9485874d744de5d9a3a5da17c25c648b5